src/tests/BasePluginTests.ts

```diff
@@ -206,10 +203,71 @@ describe("Data File helper Tests", function() {
       });
   });
 
-  it("ConfigurationFile: Should call the proper gateway URL", function(
-    done
-  ) {
+  it("DataFile: Should use cache second time a file is called and ", function(done) {
+
+    const secondFile = new Buffer("Hola");
+    const fakeDataFileURI = "mics://fake_dir/fake_file";
+    const method = "GET";
+    const dataFileGatewayURI = "/v1/data_file/data";
+    const plugin = new MyFakePlugin(false);
+    const rpMockup = sinon
+      .stub()
+      .onFirstCall()
+      .returns(Promise.resolve(fakeDataFile))
+      .onSecondCall()
+      .throws('this should fail')
```

Comment:
I think that we should try to have a better mockup here to specifically reproduce the `statusCode=301` situation.